### DISCUSSION TITLE
Build error with Unexpected token when importing react-native/tvos-types.d.ts

### DISCUSSION THREAD

**Discussion by matias-sandell** [react-native-tvos/react-native-tvos#474] [Q&A] [answered]
Trying out the new https://www.npmjs.com/package/react-native-tvos/v/0.69.8-0rc3 release and getting a build error when a TypeScript component imports `react-native/tvos-types.d`.

```
error SyntaxError: ......../node_modules/react-native/tvos-types.d.ts: Unexpected token (115:21)

  113 |   export type FocusGuideMethods = {
  114 |     setDestinations: (
> 115 |       destinations: (?React.ElementRef<HostComponent<unknown>>)[],
      |                      ^
  116 |     ) => void;
  117 |   }
  118 |.

FAILURE: Build failed with an exception.
```

<img width="981" alt="Screenshot 2023-03-16 at 11 16 03" src="https://user-images.githubusercontent.com/437760/225587679-0693afe5-66e4-4fc5-bc06-80f5b00662a9.png">

Workarounds
- Removing the `react-native/tvos-types.d` import from the .tsx file makes the build work again (and the app works when building for production, but not when running it through the react native dev server)
- Removing the `?` from the type definition also makes the app work through the dev server.

It's probably something with my setup but any hints as to what the underlying problem might be will be greatly appreciated! 🙌

The setDestinations type definition was introduced in https://github.com/react-native-tvos/react-native-tvos/commit/7143d7877ec432337f36693fba75f611178a96e8#diff-e14263f50370e82b314d0b19b9a21e70e0a0948798e28872703fdd1093b3610cR111

**Comment by yusufyildirim** [comment, marked as answer]
Thanks for reporting! You can track the progress under #476. Fix is ready #477.
